dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/csv/DefaultCsvImportService.java

```diff
@@ -700,25 +691,18 @@ private void setOptionGroupsFromCsv( CsvReader reader, Metadata metadata )
                 }
 
                 OptionSet optionSet = new OptionSet();
-                optionSet.setUid( getSafe( values, 4, null, 11 ) );
+                optionSet.setUid( getSafe( values, 4, 11 ) );
 
                 if ( optionSet.getUid() == null )
                 {
                     continue;
                 }
 
                 OptionSet persistedOptionSet = optionSet.getUid() != null
-                    ? mapOptionSet.get( optionSet.getUid(), () -> {
-                        OptionSet persistedOS = optionService.getOptionSet( optionSet.getUid() );
-                        persistedOS.getOptions();
-                        return persistedOS;
-                    } )
-                    : mapOptionSet.get( optionSet.getCode(), () -> {
-                        OptionSet persistedOS = optionService.getOptionSetByCode( optionSet.getUid() );
```

Comment:
Note the `getUid()` being wrongly used here instead of `getCode()`.

Reply:
Interesting. Could you backport this bugfix? It might be safer to backport just this line vs the refactor.

Reply:
Backport done for `2.36`, `2.35` and `2.34`